src/librustc_typeck/check/method/suggest.rs

```diff
@@ -569,6 +569,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     });
                     self.suggest_use_candidates(&mut err, help, candidates);
                 }
+                if let ty::Ref(region, t_type, mutability) = rcvr_ty.kind {
+                    let trait_type = match mutability {
```

Comment:
This is duplicated here from the file above -- could we introduce `self.tcx.mk_ref(region, t_type, mutability)` and dedup?